flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogWriter.java

```diff
@@ -329,14 +343,28 @@ public void confirm(SequenceNumber from, SequenceNumber to) {
         uploaded.subMap(from, to).values().stream()
                 .map(UploadResult::getStreamStateHandle)
                 .forEach(changelogRegistry::stopTracking);
+
+        if (localRecoveryConfig.isLocalRecoveryEnabled()) {
+            LocalStateRegistry localStateRegistry =
+                    localRecoveryConfig
+                            .getLocalStateRegistry()
+                            .orElseThrow(LocalRecoveryConfig.localRecoveryNotEnabled());
+
+            // transfer the control of localHandle to localStateRegistry.
+            for (UploadResult result : uploaded.subMap(from, to).values()) {
+                changelogRegistry.stopTracking(result.localStreamHandle);
+                localStateRegistry.register(result.localStreamHandle, checkpointId);
+            }
+            localStateRegistry.unRegister(checkpointId);
```

Comment:
Just to clarify: we only retain local state for the last checkpoint (num retained checkpoint is ignored). Same as the current local recovery, right?

Reply:
👍Right, we only retain local state for the last one checkpoint.

Reply:
Thanks!